// moi-ui-lib/dist/index.es.js
import D from "react";
var j = { exports: {} }, b = {};
/**
 * @license React
 * react-jsx-runtime.production.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var L;
function ne() {
  if (L) return b;
  L = 1;
  var s = Symbol.for("react.transitional.element"), l = Symbol.for("react.fragment");
  function f(m, i, c) {
    var p = null;
    if (c !== void 0 && (p = "" + c), i.key !== void 0 && (p = "" + i.key), "key" in i) {
      c = {};
      for (var v in i)
        v !== "key" && (c[v] = i[v]);
    } else c = i;
    return i = c.ref, {
      $$typeof: s,
      type: m,
      key: p,
      ref: i !== void 0 ? i : null,
      props: c
    };
  }
  return b.Fragment = l, b.jsx = f, b.jsxs = f, b;
}
var g = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var M;
function oe() {
  return M || (M = 1, process.env.NODE_ENV !== "production" && (function() {
    function s(e) {
      if (e == null) return null;
      if (typeof e == "function")
        return e.$$typeof === ee ? null : e.displayName || e.name || null;
      if (typeof e == "string") return e;
      switch (e) {
        case E:
          return "Fragment";
        case H:
          return "Profiler";
        case J:
          return "StrictMode";
        case X:
          return "Suspense";
        case Z:
          return "SuspenseList";
        case K:
          return "Activity";
      }
      if (typeof e == "object")
        switch (typeof e.tag == "number" && console.error(
          "Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."
        ), e.$$typeof) {
          case W:
            return "Portal";
          case B:
            return (e.displayName || "Context") + ".Provider";
          case q:
            return (e._context.displayName || "Context") + ".Consumer";
          case G:
            var t = e.render;
            return e = e.displayName, e || (e = t.displayName || t.name || "", e = e !== "" ? "ForwardRef(" + e + ")" : "ForwardRef"), e;
          case Q:
            return t = e.displayName || null, t !== null ? t : s(e.type) || "Memo";
          case P:
            t = e._payload, e = e._init;
            try {
              return s(e(t));
            } catch {
            }
        }
      return null;
    }
    function l(e) {
      return "" + e;
    }
    function f(e) {
      try {
        l(e);
        var t = !1;
      } catch {
        t = !0;
      }
      if (t) {
        t = console;
        var n = t.error, o = typeof Symbol == "function" && Symbol.toStringTag && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return n.call(
          t,
          "The provided key is an unsupported type %s. This value must be coerced to a string before using it here.",
          o
        ), l(e);
      }
    }
    function m(e) {
      if (e === E) return "<>";
      if (typeof e == "object" && e !== null && e.$$typeof === P)
        return "<...>";
      try {
        var t = s(e);
        return t ? "<" + t + ">" : "<...>";
      } catch {
        return "<...>";
      }
    }
    function i() {
      var e = R.A;
      return e === null ? null : e.getOwner();
    }
    function c() {
      return Error("react-stack-top-frame");
    }
    function p(e) {
      if (N.call(e, "key")) {
        var t = Object.getOwnPropertyDescriptor(e, "key").get;
        if (t && t.isReactWarning) return !1;
      }
      return e.key !== void 0;
    }
    function v(e, t) {
      function n() {
        C || (C = !0, console.error(
          "%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://react.dev/link/special-props)",
          t
        ));
      }
      n.isReactWarning = !0, Object.defineProperty(e, "key", {
        get: n,
        configurable: !0
      });
    }
    function $() {
      var e = s(this.type);
      return Y[e] || (Y[e] = !0, console.error(
        "Accessing element.ref was removed in React 19. ref is now a regular prop. It will be removed from the JSX Element type in a future release."
      )), e = this.props.ref, e !== void 0 ? e : null;
    }
    function U(e, t, n, o, d, u, k, T) {
      return n = u.ref, e = {
        $$typeof: O,
        type: e,
        key: t,
        props: u,
        _owner: d
      }, (n !== void 0 ? n : null) !== null ? Object.defineProperty(e, "ref", {
        enumerable: !1,
        get: $
      }) : Object.defineProperty(e, "ref", { enumerable: !1, value: null }), e._store = {}, Object.defineProperty(e._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: 0
      }), Object.defineProperty(e, "_debugInfo", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: null
      }), Object.defineProperty(e, "_debugStack", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: k
      }), Object.defineProperty(e, "_debugTask", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: T
      }), Object.freeze && (Object.freeze(e.props), Object.freeze(e)), e;
    }
    function y(e, t, n, o, d, u, k, T) {
      var a = t.children;
      if (a !== void 0)
        if (o)
          if (re(a)) {
            for (o = 0; o < a.length; o++)
              S(a[o]);
            Object.freeze && Object.freeze(a);
          } else
            console.error(
              "React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead."
            );
        else S(a);
      if (N.call(t, "key")) {
        a = s(e);
        var h = Object.keys(t).filter(function(te) {
          return te !== "key";
        });
        o = 0 < h.length ? "{key: someKey, " + h.join(": ..., ") + ": ...}" : "{key: someKey}", I[a + o] || (h = 0 < h.length ? "{" + h.join(": ..., ") + ": ...}" : "{}", console.error(
          `A props object containing a "key" prop is being spread into JSX:
  let props = %s;
  <%s {...props} />
React keys must be passed directly to JSX without using spread:
  let props = %s;
  <%s key={someKey} {...props} />`,
          o,
          a,
          h,
          a
        ), I[a + o] = !0);
      }
      if (a = null, n !== void 0 && (f(n), a = "" + n), p(t) && (f(t.key), a = "" + t.key), "key" in t) {
        n = {};
        for (var w in t)
          w !== "key" && (n[w] = t[w]);
      } else n = t;
      return a && v(
        n,
        typeof e == "function" ? e.displayName || e.name || "Unknown" : e
      ), U(
        e,
        a,
        u,
        d,
        i(),
        n,
        k,
        T
      );
    }
    function S(e) {
      typeof e == "object" && e !== null && e.$$typeof === O && e._store && (e._store.validated = 1);
    }
    var x = D, O = Symbol.for("react.transitional.element"), W = Symbol.for("react.portal"), E = Symbol.for("react.fragment"), J = Symbol.for("react.strict_mode"), H = Symbol.for("react.profiler"), q = Symbol.for("react.consumer"), B = Symbol.for("react.context"), G = Symbol.for("react.forward_ref"), X = Symbol.for("react.suspense"), Z = Symbol.for("react.suspense_list"), Q = Symbol.for("react.memo"), P = Symbol.for("react.lazy"), K = Symbol.for("react.activity"), ee = Symbol.for("react.client.reference"), R = x.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE, N = Object.prototype.hasOwnProperty, re = Array.isArray, _ = console.createTask ? console.createTask : function() {
      return null;
    };
    x = {
      react_stack_bottom_frame: function(e) {
        return e();
      }
    };
    var C, Y = {}, F = x.react_stack_bottom_frame.bind(
      x,
      c
    )(), z = _(m(c)), I = {};
    g.Fragment = E, g.jsx = function(e, t, n, o, d) {
      var u = 1e4 > R.recentlyCreatedOwnerStacks++;
      return y(
        e,
        t,
        n,
        !1,
        o,
        d,
        u ? Error("react-stack-top-frame") : F,
        u ? _(m(e)) : z
      );
    }, g.jsxs = function(e, t, n, o, d) {
      var u = 1e4 > R.recentlyCreatedOwnerStacks++;
      return y(
        e,
        t,
        n,
        !0,
        o,
        d,
        u ? Error("react-stack-top-frame") : F,
        u ? _(m(e)) : z
      );
    };
  })()), g;
}
var V;
function ae() {
  return V || (V = 1, process.env.NODE_ENV === "production" ? j.exports = ne() : j.exports = oe()), j.exports;
}
var r = ae();
const A = ({
  label: s,
  ...l
}) => /* @__PURE__ */ r.jsx(
  "button",
  {
    type: "button",
    className: "bg-primary-500",
    ...l,
    children: s
  }
), se = ({ user: s, onLogin: l, onLogout: f, onCreateAccount: m }) => /* @__PURE__ */ r.jsx("header", { children: /* @__PURE__ */ r.jsxs("div", { className: "storybook-header", children: [
  /* @__PURE__ */ r.jsxs("div", { children: [
    /* @__PURE__ */ r.jsx("svg", { width: "32", height: "32", viewBox: "0 0 32 32", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ r.jsxs("g", { fill: "none", fillRule: "evenodd", children: [
      /* @__PURE__ */ r.jsx(
        "path",
        {
          d: "M10 0h12a10 10 0 0110 10v12a10 10 0 01-10 10H10A10 10 0 010 22V10A10 10 0 0110 0z",
          fill: "#FFF"
        }
      ),
      /* @__PURE__ */ r.jsx(
        "path",
        {
          d: "M5.3 10.6l10.4 6v11.1l-10.4-6v-11zm11.4-6.2l9.7 5.5-9.7 5.6V4.4z",
          fill: "#555AB9"
        }
      ),
      /* @__PURE__ */ r.jsx(
        "path",
        {
          d: "M27.2 10.6v11.2l-10.5 6V16.5l10.5-6zM15.7 4.4v11L6 10l9.7-5.5z",
          fill: "#91BAF8"
        }
      )
    ] }) }),
    /* @__PURE__ */ r.jsx("h1", { children: "Acme" })
  ] }),
  /* @__PURE__ */ r.jsx("div", { children: s ? /* @__PURE__ */ r.jsxs(r.Fragment, { children: [
    /* @__PURE__ */ r.jsxs("span", { className: "welcome", children: [
      "Welcome, ",
      /* @__PURE__ */ r.jsx("b", { children: s.name }),
      "!"
    ] }),
    /* @__PURE__ */ r.jsx(A, { onClick: f, label: "Log out" })
  ] }) : /* @__PURE__ */ r.jsxs(r.Fragment, { children: [
    /* @__PURE__ */ r.jsx(A, { onClick: l, label: "Log in" }),
    /* @__PURE__ */ r.jsx(A, { onClick: m, label: "Sign up" })
  ] }) })
] }) }), ie = () => {
  const [s, l] = D.useState();
  return /* @__PURE__ */ r.jsxs("article", { children: [
    /* @__PURE__ */ r.jsx(
      se,
      {
        user: s,
        onLogin: () => l({ name: "Jane Doe" }),
        onLogout: () => l(void 0),
        onCreateAccount: () => l({ name: "Jane Doe" })
      }
    ),
    /* @__PURE__ */ r.jsxs("section", { className: "storybook-page", children: [
      /* @__PURE__ */ r.jsx("h2", { children: "Pages in Storybook" }),
      /* @__PURE__ */ r.jsxs("p", { children: [
        "We recommend building UIs with a",
        " ",
        /* @__PURE__ */ r.jsx("a", { href: "https://componentdriven.org", target: "_blank", rel: "noopener noreferrer", children: /* @__PURE__ */ r.jsx("strong", { children: "component-driven" }) }),
        " ",
        "process starting with atomic components and ending with pages."
      ] }),
      /* @__PURE__ */ r.jsx("p", { children: "Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:" }),
      /* @__PURE__ */ r.jsxs("ul", { children: [
        /* @__PURE__ */ r.jsx("li", { children: 'Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories' }),
        /* @__PURE__ */ r.jsx("li", { children: "Assemble data in the page component from your services. You can mock these services out using Storybook." })
      ] }),
      /* @__PURE__ */ r.jsxs("p", { children: [
        "Get a guided tutorial on component-driven development at",
        " ",
        /* @__PURE__ */ r.jsx("a", { href: "https://storybook.js.org/tutorials/", target: "_blank", rel: "noopener noreferrer", children: "Storybook tutorials" }),
        ". Read more in the",
        " ",
        /* @__PURE__ */ r.jsx("a", { href: "https://storybook.js.org/docs", target: "_blank", rel: "noopener noreferrer", children: "docs" }),
        "."
      ] }),
      /* @__PURE__ */ r.jsxs("div", { className: "tip-wrapper", children: [
        /* @__PURE__ */ r.jsx("span", { className: "tip", children: "Tip" }),
        " Adjust the width of the canvas with the",
        " ",
        /* @__PURE__ */ r.jsx("svg", { width: "10", height: "10", viewBox: "0 0 12 12", xmlns: "http://www.w3.org/2000/svg", children: /* @__PURE__ */ r.jsx("g", { fill: "none", fillRule: "evenodd", children: /* @__PURE__ */ r.jsx(
          "path",
          {
            d: "M1.5 5.2h4.8c.3 0 .5.2.5.4v5.1c-.1.2-.3.3-.4.3H1.4a.5.5 0 01-.5-.4V5.7c0-.3.2-.5.5-.5zm0-2.1h6.9c.3 0 .5.2.5.4v7a.5.5 0 01-1 0V4H1.5a.5.5 0 010-1zm0-2.1h9c.3 0 .5.2.5.4v9.1a.5.5 0 01-1 0V2H1.5a.5.5 0 010-1zm4.3 5.2H2V10h3.8V6.2z",
            id: "a",
            fill: "#999"
          }
        ) }) }),
        "Viewports addon in the toolbar"
      ] })
    ] })
  ] });
};
export {
  A as Button,
  se as Header,
  ie as Page
};
